src/matlab2tikz.m

```diff
@@ -1911,7 +1947,20 @@ function assertRegularAxes(handle)
     matlabDefaultLineWidth = 0.5;
     if m2t.args.strict ...
             || ~abs(lineWidth-matlabDefaultLineWidth) <= m2t.tol
-        lineOpts = opts_add(lineOpts, 'line width', sprintf('%.1fpt', lineWidth));
+        if m2t.args.semanticLineWidth ...
+                && ismember(lineWidth, [m2t.args.semanticLineWidthDefinition{:,2}])
+            semStrID = lineWidth == [m2t.args.semanticLineWidthDefinition{:,2}];
+            lineOpts = opts_add(lineOpts, m2t.args.semanticLineWidthDefinition{semStrID,1}, []);
+        else
+            lineOpts = opts_add(lineOpts, 'line width', sprintf('%.1fpt', lineWidth));
+        end
+    else % lineWidth = matlabDefaultLineWidth
+        if m2t.args.semanticLineWidth
```

Comment:
Here the check whether there is a matching string is missing. Please add a separate if clause and put a warning into the else (not matching string) clause that we fallback to points